pkg/provider/acme/local_store.go

```diff
@@ -56,49 +63,40 @@ func (s *LocalStore) get() (*StoredData, error) {
 			}
 
 			if len(file) > 0 {
-				if err := json.Unmarshal(file, s.storedData); err != nil {
-					return nil, err
-				}
-			}
-
-			// Check if ACME Account is in ACME V1 format
-			if s.storedData.Account != nil && s.storedData.Account.Registration != nil {
-				isOldRegistration, err := regexp.MatchString(RegistrationURLPathV1Regexp, s.storedData.Account.Registration.URI)
-				if err != nil {
+				if err := json.Unmarshal(file, &s.storedData); err != nil {
 					return nil, err
 				}
-				if isOldRegistration {
-					logger.Debug("Reseting ACME account.")
-					s.storedData.Account = nil
-					s.SaveDataChan <- s.storedData
-				}
 			}
 
 			// Delete all certificates with no value
-			var certificates []*Certificate
-			for _, certificate := range s.storedData.Certificates {
-				if len(certificate.Certificate) == 0 || len(certificate.Key) == 0 {
-					logger.Debugf("Deleting empty certificate %v for %v", certificate, certificate.Domain.ToStrArray())
-					continue
+			var certificates []*CertAndStore
+			for _, storedData := range s.storedData {
+				for _, certificate := range storedData.Certificates {
+					if len(certificate.Certificate.Certificate) == 0 || len(certificate.Key) == 0 {
+						logger.Debugf("Deleting empty certificate %v for %v", certificate, certificate.Domain.ToStrArray())
+						continue
+					}
+					certificates = append(certificates, certificate)
+				}
+				if len(certificates) < len(storedData.Certificates) {
+					storedData.Certificates = certificates
+					s.saveDataChan <- s.storedData
 				}
-				certificates = append(certificates, certificate)
-			}
-
-			if len(certificates) < len(s.storedData.Certificates) {
-				s.storedData.Certificates = certificates
-				s.SaveDataChan <- s.storedData
 			}
 		}
 	}
 
-	return s.storedData, nil
+	if s.storedData[resolverName] == nil {
+		s.storedData[resolverName] = &StoredData{}
```

Comment:
why do we do that? do we have callers of get() relying on it never returning nil? if yes, why?